src/gui/addnewtorrentdialog.cpp

```diff
@@ -621,32 +636,63 @@ void AddNewTorrentDialog::displayContentTreeMenu(const QPoint &)
     }
 }
 
+void AddNewTorrentDialog::setAddTorrentParams(const BitTorrent::AddTorrentParams &inParams)
+{
+    torrentParams = inParams;
+    
+    if (!torrentParams.savePath.isEmpty())
+        setSavePath(torrentParams.savePath);
+    
+    if (!torrentParams.category.isEmpty()) {
+        int index = ui->categoryComboBox->findText(torrentParams.category);
+        
+        if (index != -1) {
+            ui->categoryComboBox->setCurrentIndex(index);
+        }
+        else {
+            ui->categoryComboBox->addItem(torrentParams.category);
+            ui->categoryComboBox->model()->sort(0);
+            ui->categoryComboBox->setCurrentIndex(ui->categoryComboBox->findText(torrentParams.category));
+        }
+    }
+    
+    if (torrentParams.addPaused == TriStateBool::True) {
+        ui->startTorrentCheckBox->setChecked(false);
+    }
+    else if (torrentParams.addPaused == TriStateBool::False) {
+        ui->startTorrentCheckBox->setChecked(true);
+    }
```

Comment:
If `Undefined` then we leave the checkbox as is, since we fall back on the application setting. So nothing to do in that case, hence no `else` clause.